community/kernel/src/main/java/org/neo4j/kernel/impl/newapi/DefaultNodeValueIndexCursor.java

```diff
@@ -75,19 +75,23 @@ public void initialize( SchemaIndexDescriptor descriptor, IndexProgressor progre
             seekQuery( descriptor, query );
             break;
 
-        case stringSuffix:
-        case stringContains:
         case exists:
             scanQuery( descriptor );
             break;
 
         case range:
             assert query.length == 1;
-            rangeQuery( descriptor, (IndexQuery.RangePredicate) query[0] );
+            rangeQuery( descriptor, (IndexQuery.RangePredicate) firstPredicate );
             break;
 
         case stringPrefix:
-            prefixQuery( descriptor, (IndexQuery.StringPrefixPredicate) query[0] );
+            prefixQuery( descriptor, (IndexQuery.StringPrefixPredicate) firstPredicate );
```

Comment:
looks like  assert query.length == 1; is missing here

Reply:
yupp :)